Modifying test_target_private.c to use headerfile (Issue #10)

tmh97 opened the pull request:
Making style changes and including macro usage. Still need to add description and do some more testing, currently passing on GCC and Clang.

josemonsalve2 commented on tests/4.5/target/test_target_private.c:
```diff
@@ -4,53 +4,53 @@
 //
 //Testing private clause with target directive
```
Improve the description.

josemonsalve2 commented on tests/4.5/target/test_target_private.c:
```diff
@@ -4,53 +4,53 @@
-    for (j=0; j<N; j++){
-      if(compute_array[i][j] != 100){
-        errors += 1;
+  for (i = 0; i < 4; i++){ 
```
The actual number of threads may be less than 4. So we would need to get `omp_get_num_threads` in the parallel region, and use that number in this verification code.

tmh97 replied:
Is it valid to replace 44 with `for (i = 0; i < omp_get_thread_num(); i++)`

josemonsalve2 replied:
No because it is already outside of the parallel region, so the number of threads is always 1. You have to use a variable to probe the number of threads in the parallel region. 

In general, `omp_get_num_threads()` will give you the number of threads in the current execution context where it is called.

josemonsalve2 commented on tests/4.5/target/test_target_private.c:
```diff
@@ -4,53 +4,53 @@
+    for (j = 0; j < N; j++){
+      OMPVV_TEST_AND_SET_VERBOSE(errors, (compute_array[i][j] != 100));
+      if (compute_array[i][j] != 100) 
         printf("compute_array[%d][%d] = %d\n",i,j,compute_array[i][j]);
```
Use `OMPVV_ERROR_IF(compute_array[i][j] != 100, "compute_array[%d][%d] = %d\n",i,j,compute_array[i][j])` instead

tmh97 replied:
Taking care of these recommendations now, thank you!

tmh97 commented:
@josemonsalve2 I updated header, added a probe for the real number of threads, and added OMPVV_ERROR_IF macro at the end instead of traditional printf setup. Let me know how these changes are looking, thanks again for looking this over.

tmh97 commented:
Testing again before merging, passes on gcc and clang.
